internal/orchestration/orchestration_test.go

```diff
@@ -77,85 +79,83 @@ func makeClientOpts() *orchestrationpb.ClientOpts {
-	t.Run("Fast-HotStuff+EDDSA", func(t *testing.T) { run(t, replicaOpts) })
-	replicaOpts = makeReplicaOpts("fasthotstuff", "bls12", "", nil)
-	t.Run("Fast-HotStuff+BLS12", func(t *testing.T) { run(t, replicaOpts) })
-	replicaOpts = makeReplicaOpts("simplehotstuff", "ecdsa", "", nil)
-	t.Run("Simple-HotStuff+ECDSA", func(t *testing.T) { run(t, replicaOpts) })
-	replicaOpts = makeReplicaOpts("simplehotstuff", "eddsa", "", nil)
-	t.Run("Simple-HotStuff+EDDSA", func(t *testing.T) { run(t, replicaOpts) })
-	replicaOpts = makeReplicaOpts("simplehotstuff", "bls12", "", nil)
-	t.Run("Simple-HotStuff+BLS12", func(t *testing.T) { run(t, replicaOpts) })
-
-	// byzantine
-	replicaOpts = makeReplicaOpts("chainedhotstuff", "ecdsa", "fork:1", nil)
-	t.Run("ChainedHotStuff+Fork", func(t *testing.T) { run(t, replicaOpts) })
-	replicaOpts = makeReplicaOpts("chainedhotstuff", "ecdsa", "silence:1", nil)
-	t.Run("ChainedHotStuff+Silence", func(t *testing.T) { run(t, replicaOpts) })
+	c := make(chan error)
+	go func() {
+		c <- worker.Run()
+	}()
 
+	err = experiment.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = <-c
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestOrchestration(t *testing.T) {
+	tests := []struct {
+		consensus  string
+		crypto     string
+		byzantine  string
+		mods       []string
+		randomTree bool
+	}{
+		{consensus: "chainedhotstuff", crypto: "ecdsa", byzantine: "", mods: nil},
+		{consensus: "chainedhotstuff", crypto: "eddsa", byzantine: "", mods: nil},
+		{consensus: "chainedhotstuff", crypto: "bls12", byzantine: "", mods: nil},
+		{consensus: "fasthotstuff", crypto: "ecdsa", byzantine: "", mods: nil},
+		{consensus: "fasthotstuff", crypto: "eddsa", byzantine: "", mods: nil},
+		{consensus: "fasthotstuff", crypto: "bls12", byzantine: "", mods: nil},
+		{consensus: "simplehotstuff", crypto: "ecdsa", byzantine: "", mods: nil},
+		{consensus: "simplehotstuff", crypto: "eddsa", byzantine: "", mods: nil},
+		{consensus: "simplehotstuff", crypto: "bls12", byzantine: "", mods: nil},
+		{consensus: "chainedhotstuff", crypto: "ecdsa", byzantine: "fork:1", mods: nil},
+		{consensus: "chainedhotstuff", crypto: "ecdsa", byzantine: "silence:1", mods: nil},
+		{consensus: "chainedhotstuff", crypto: "ecdsa", byzantine: "", mods: []string{"kauri"}},
+		{consensus: "chainedhotstuff", crypto: "bls12", byzantine: "", mods: []string{"kauri"}},
+		{consensus: "chainedhotstuff", crypto: "ecdsa", byzantine: "", mods: []string{"kauri"}, randomTree: true},
+		{consensus: "chainedhotstuff", crypto: "bls12", byzantine: "", mods: []string{"kauri"}, randomTree: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(test.Name([]string{"consensus", "crypto", "byzantine", "mods"}, tt.consensus, tt.crypto, tt.byzantine, tt.mods), func(t *testing.T) {
+			var replicaOpts *orchestrationpb.ReplicaOpts
+			if slices.Contains(tt.mods, "kauri") {
+				replicaOpts = makeTreeReplicaOpts(tt.consensus, tt.crypto, tt.mods, 7, 2, tt.randomTree)
```

Comment:
We should make 7, 2 be part of the `tests` struct, and leave it unspecified (0) for the non-kauri tests.